util/file_reader_writer.h

```diff
@@ -109,6 +112,8 @@ class RandomAccessFileReader {
 
   RandomAccessFile* file() { return file_.get(); }
 
+  std::string file_name() const { return file_name_; }
```

Comment:
How about adding `file_name()` method to `RandomAccessFile` and let `RandomAccessFileReader::file_name()` just relay the result?